### PR TITLE
Added tests, fixed bugs, other enhancements

### PR DESCRIPTION
**Pull request by vmw-loginsight**
Testing

* Added tox with pytest, coverage, flakes, and pep8
* Added travis-ci and coveralls
* Added test coverage for init, bugzilla, and slack
* Updated readme
* Added contributing document

init

* parseLI did not properly pass info in payload
* parsevROps did not work for versions less than 6.2
* callapi did not treat check as boolean
* Improved error handling

Bugzilla

* Added support for product/component/version without token
* Prefer product/component/version to BUGZILLA parameters
* Properly catch and report errors
* Added support for landfill.bugzilla.org

Slack

* Refactored field processing
* More verification for SLACKURL
* Ensure icon is added to all message types
* Properly handle test messages from non-test messages

### CODE REVIEW

File: loginsightwebhookdemo/bugzilla.py [hunk 1]
```diff
@@ -81,4 +88,10 @@ def bugzilla(ALERTID=None, TOKEN=None, PRODUCT=None, COMPONENT=None, VERSION=Non
             "summary" : a['AlertName'],
             "description": a['info'],
         }
-        return callapi(BUGZILLAURL + '/rest/bug?' + auth, 'post', json.dumps(payload), headers, None, check = VERIFY)
+        # op_sys and rep_platform may not be needed, but are to test on landfill
+        if "landfill.bugzilla.org" in BUGZILLAURL:
```
Comment (alanjcastonguay) [inline loginsightwebhookdemo/bugzilla.py]: That's a matching a substring anywhere in the `BUGZILLAURL`, which is not necessarily the hostname involved. It's a tiny edge-case, but it a bit wider of a net than intended.

File: tests/bugzilla_test.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    rsp = client.post('/endpoint/bugzilla/' + TOKEN, data=payload, content_type="application/json")
+    assert rsp.status == '200 OK'
+    rsp = client.post('/endpoint/bugzilla', data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+    rsp = client.post('/endpoint/bugzilla/' + PRODUCT + '/' + COMPONENT + '/' + VERSION, data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+
+
+def test_bugzilla_wrongtoken():
+    loginsightwebhookdemo.bugzilla.BUGZILLAPASS = 'L0g1ns1ght!'
```
Comment (alanjcastonguay) [inline tests/bugzilla_test.py]: Did you intend to push this password?

File: tests/bugzilla_test.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        {
+            "text": "hello world 1",
+            "timestamp": 1451940578545,
+            "fields": []
+        }
+    ]
+}
+
+URL = 'https://landfill.bugzilla.org/bugzilla-5.0-branch/'
+TOKEN = 'BR8zMD8ywOJSAgRnjylr4G3pa7Zx5zKE6U3xYl76'
```
Comment (alanjcastonguay) [inline tests/bugzilla_test.py]: Should this `TOKEN` be here?

File: tests/bugzilla_test.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+
+
+import json
+import uuid
+
+import loginsightwebhookdemo
+import loginsightwebhookdemo.bugzilla
```
Comment (alanjcastonguay) [inline tests/bugzilla_test.py]: Consider using `requests-mock` to avoid the need to round-trip to a real server.

File: loginsightwebhookdemo/bugzilla.py (hunk 1 above)
Comment (vmw-loginsight) [inline loginsightwebhookdemo/bugzilla.py]: Indeed -- I consider this a poor's man approach to validating the URL. Trying to prevent the user from doing something undesired with a cheap check.

File: tests/bugzilla_test.py (hunk 3 above)
Comment (vmw-loginsight) [inline tests/bugzilla_test.py]: I opted for yes -- this is how it tests with an external server. I understand someone could change the credentials, however anyone can register for an account on this server today so I am hoping the risk is low and if problems arise, the fix is trivial.

File: tests/bugzilla_test.py (hunk 2 above)
Comment (vmw-loginsight) [inline tests/bugzilla_test.py]: Same as token -- goal is to test code + provide information tester would need in order to validate.

File: loginsightwebhookdemo/bugzilla.py (hunk 1 above)
Comment (alanjcastonguay) [inline loginsightwebhookdemo/bugzilla.py]: Ok

File: tests/bugzilla_test.py (hunk 3 above)
Comment (alanjcastonguay) [inline tests/bugzilla_test.py]: Ok

File: tests/bugzilla_test.py (hunk 2 above)
Comment (alanjcastonguay) [inline tests/bugzilla_test.py]: Ok
